### PR TITLE
I6902 extracting out installError component

### LINKED ISSUE #6902: Extract out renderInstallError into its own component
Now that we've used this a few times, we should make this into its own component.

QA instructions: Please make sure install error notices still appear

### PR DESCRIPTION
**Pull request by rebmullin**
fixes #6902

### CODE REVIEW

**Comment by willdurand** [comment]
> QA instructions: Please make sure install error notices still appear

This should be added to the original issue.

File: src/amo/components/AddonInstallError/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  i18n: I18nType,
+|};
+
+export const AddonInstallErrorBase = ({ error, i18n }: InternalProps) => {
+  if (!error) {
+    return null;
+  }
+
+  return (
+    <Notice className="Addon-header-install-error" type="error">
```
Comment (willdurand) [inline src/amo/components/AddonInstallError/index.js]: The class name should be adjusted to match the component's name, a.k.a. `AddonInstallError`.

File: src/amo/components/AddonInstallError/index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      {getErrorMessage({ i18n, error })}
+    </Notice>
+  );
+};
+
+const AddonInstallError: React.ComponentType<InternalProps> = translate()(
```
Comment (willdurand) [inline src/amo/components/AddonInstallError/index.js]:
```suggestion
const AddonInstallError: React.ComponentType<Props> = translate()(
```

The type `React.ComponentType<...>` describes the interface/public API of the `AddonInstallError` component, so we should pass `Props` and not `InternalProps`.

File: src/amo/components/GuidesAddonCard/index.js [hunk 3]
```diff
@@ -104,7 +88,7 @@ export class GuidesAddonCardBase extends React.Component<InternalProps> {
 
     return addon ? (
       <Card>
-        {this.renderInstallError()}
+        <AddonInstallError i18n={i18n} error={this.props.installError} />
```
Comment (willdurand) [inline src/amo/components/GuidesAddonCard/index.js]: You should not pass `i18n` here, it is injected in the `AddonInstallError` internally (via the `translate()` HOC).

File: tests/unit/amo/components/TestAddonInstallError.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+} from 'amo/components/AddonInstallError';
+import { fakeI18n, shallowUntilTarget } from 'tests/unit/helpers';
+import { FATAL_ERROR } from 'core/constants';
+
+function renderProps({ i18n = fakeI18n(), ...customProps } = {}) {
+  return {
+    i18n,
+    ...customProps,
+  };
+}
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonInstallError.js]: Please put this function into the top-level `describe` block.

File: tests/unit/amo/components/TestAddonInstallError.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+function render(otherProps = {}) {
+  const props = renderProps({ ...otherProps });
+  return shallowUntilTarget(
+    <AddonInstallError {...props} />,
+    AddonInstallErrorBase,
+  );
+}
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonInstallError.js]: Please put this function into the top-level `describe` block.

Comment (willdurand) [reply tests/unit/amo/components/TestAddonInstallError.js]: Also you could use the short syntax for functions: `() => {}`

File: tests/unit/amo/components/TestAddonInstallError.js (hunk 4 above)
Comment (willdurand) [reply tests/unit/amo/components/TestAddonInstallError.js]: I don't think this function is needed actually.

File: tests/unit/amo/components/TestAddonInstallError.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+function renderProps({ i18n = fakeI18n(), ...customProps } = {}) {
+  return {
+    i18n,
+    ...customProps,
+  };
+}
+
+function render(otherProps = {}) {
+  const props = renderProps({ ...otherProps });
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonInstallError.js]: You could just do:

```
const render = (props = {}) => {
  const allProps = {
    error: null,
    ...props,
  };

  return shallowUntilTarget(<AddonInstallError i18n={fakeI18n()} {...props} />, ...);
};
```

File: tests/unit/amo/components/TestAddonInstallError.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+    expect(root.find('.Addon-header-install-error')).toHaveLength(0);
+  });
+
+  it('renders an install error if there is one', () => {
+    const error = FATAL_ERROR;
+    const root = render({ error });
+
+    expect(root.find('.Addon-header-install-error')).toHaveLength(1);
+    expect(root.find('.Addon-header-install-error')).toHaveProp(
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonInstallError.js]: Can you use `toHaveText` by chance?

File: tests/unit/amo/components/TestGuidesAddonCard.js [hunk 8]
```diff
@@ -191,36 +185,9 @@ describe(__filename, () => {
...
-
-    expect(root.find('.Addon-header-install-error')).toHaveLength(1);
-    expect(root.find('.Addon-header-install-error')).toHaveProp(
-      'children',
-      getErrorMessage({ i18n: fakeI18n(), error }),
-    );
-  });
-
-  it('does not render an install error if there is no error', () => {
+  it('renders AddonInstallError', () => {
```
Comment (willdurand) [inline tests/unit/amo/components/TestGuidesAddonCard.js]:
```suggestion
  it('renders an AddonInstallError component', () => {
```

File: src/amo/components/AddonInstallError/index.js (hunk 1 above)
Comment (rebmullin) [reply src/amo/components/AddonInstallError/index.js]: ahh right!

File: src/amo/components/GuidesAddonCard/index.js (hunk 3 above)
Comment (rebmullin) [reply src/amo/components/GuidesAddonCard/index.js]: yep make sense

**Pull request merged**